e2e/main_test.go

```diff
@@ -99,9 +119,108 @@ func setupCluster(ctx context.Context, t testing.TB) (client.Client, *rest.Confi
 		return nil, nil, err
 	}
 	t.Log(">>> operator started successfully")
+
+	if explicitCredentialsConfigured() {
```

Comment:
IIUC you're configuring the operator config credentials even though you're forcing the prometheus DaemonSet to use them via a flag; the reason being is this is idempotent, and you can simply remove the manual DaemonSet injection later.

Reply:
Nope all good